sdk/identity/azure-identity/src/token_cache.cpp

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "private/token_cache.hpp"
+#include "private/token_cache_internals.hpp"
+
+#include <chrono>
+#include <mutex>
+
+using Azure::Identity::_detail::TokenCache;
+
+using Azure::Core::Credentials::AccessToken;
+
+decltype(TokenCache::Internals::Cache) TokenCache::Internals::Cache;
+decltype(TokenCache::Internals::CacheMutex) TokenCache::Internals::CacheMutex;
+
+#if defined(TESTING_BUILD)
+std::function<void()> TokenCache::Internals::OnBeforeCacheWriteLock;
+std::function<void()> TokenCache::Internals::OnBeforeItemWriteLock;
+#endif
+
+namespace {
+bool IsFresh(
+    std::shared_ptr<TokenCache::Internals::CacheValue> const& item,
+    std::chrono::system_clock::time_point now)
+{
+  // If cached token expires in less than MinExpiry from now, it's cached value won't be returned,
+  // newer value will be requested.
+  constexpr auto MinExpiry = std::chrono::minutes(3);
```

Comment:
Where does 3 minutes come from?

Reply:
`BearerTokenAuthenticationPolicy` has the same criteria for 2 minutes, and if it is less, then it asks credential for the new token, which will result in the code in this file being executed.
This is 1 minute more. It does not have to be - can be 5 minutes, can be 2, can be 1, so we're not taking a dependency on what `BearerTokenAuthPolicy` has, but I think it makes things simpler and would produce optimal results, in terms of executing less code.

https://github.com/Azure/azure-sdk-for-cpp/blob/fda88bf7b8389575b9d973821efc604442e10493/sdk/core/azure-core/src/http/bearer_token_authentication_policy.cpp#L22-L25

I could depend on the timeout that BearerTokenAuthPolicy has, but it would probably create an impression that it has to be 1 minute more, which is technically not.

Reply:
Every time I see a magic number (like 32 or 3), I want to know where the number came from. I'm not challenging the choice of number (I think it's fine), I'm asking that the comments explain why that number was chosen.

Reply:
Thank you, I already added it to the changes that I am going to commit soon.